PayloadPkg/OsLoader/BootParameters.c

```diff
@@ -261,6 +262,17 @@ UpdateOsParameters (
   //
   CmdFile = &LoadedImage->Image.Common.CmdFile;
 
+  if ((CurrentBootOption->BootFlags & BOOT_FLAGS_MENDER) != 0 && (CurrentBootOption->BootFlags & BOOT_FLAGS_MISC) == 0) {
+    AsciiSPrint (MenderParams, sizeof (MenderParams), " root=PARTLABEL=primary ");
+    if ((AsciiStrSize (MenderParams) + CmdFile->Size) < CMDLINE_LENGTH_MAX) {
+      AsciiStrCatS ((CHAR8 *)CmdFile->Addr, CMDLINE_LENGTH_MAX, &MenderParams[0]);
+    } else {
+      DEBUG ((DEBUG_INFO, "Unable to append Mender cmd line parameters!\n"));
+    }
+  } else if ((CurrentBootOption->BootFlags & BOOT_FLAGS_MENDER) != 0 && (CurrentBootOption->BootFlags & BOOT_FLAGS_MISC) != 0) {
+    DEBUG ((DEBUG_ERROR, "ERROR: BOOT_FLAGS_MISC & BOOT_FLAGS_MENDER cannot both be enabled!\n"));
```

Comment:
Who will set the BootFlags ?  I guess it is the config data, right ?  If so, I think this might be an un-common error.  I am not sure if it is best to use DEBUG_ERROR print or not.   Every DEBUG_ERROR print will eventually add size to the final production binary.   So I think we can only add DEBUG_ERROR print whenever it is really required.  
And again, do you consider the condition as fatal?  Should we block boot in this case ?

Reply:
Boot option config data will set these boot flag.